### PR TITLE
Changes for first Purity 6 support

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
Add support for REST 2.x authentication
Add error check for directory services failing when FA-Files is installed (current API can't deal with more than 1 DS type)
Add SMI-S (v6 required)
Add volume tags (v6 required)
Add Active DR (v6 required)
Add deleted pods
Add deleted snapshots
Remove journal info as never turned up in 1.19
Error message typo fix (purefa_volume)

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
module_utils/purefa.py
modules/purefa_info.py
modules/purefa_volume.py

### CODE REVIEW

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py [hunk 1]
```diff
@@ -754,6 +826,57 @@ def generate_pgroups_dict(array):
...
+        pods = array.list_pods(mediator=True, pending_only=True)
+        for pod in range(0, len(pods)):
+            acpod = pods[pod]['name']
+            pods_info[acpod] = {
+                'source': pods[pod]['source'],
+                'arrays': pods[pod]['arrays'],
+                'mediator': pods[pod]['mediator'],
+                'mediator_version': pods[pod]['mediator_version'],
+            }
+            if pods_info[acpod]['arrays'][0]['frozen_at']:
```
Comment (dnix101) [inline collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py:862]: Suggest to change `pods_info[acpod]['arrays'][0]['frozen_at']` to `pods_info[acpod]['arrays'][0].get('frozen_at')` for backwards compatibility with purity 5.x.

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py [hunk 2]
```diff
@@ -754,6 +826,57 @@ def generate_pgroups_dict(array):
...
+                pods_info[acpod]['link_source_count'] = pods[pod]['link_source_count']
+                pods_info[acpod]['link_target_count'] = pods[pod]['link_target_count']
+                pods_info[acpod]['promotion_status'] = pods[pod]['promotion_status']
+                pods_info[acpod]['requested_promotion_state'] = pods[pod]['requested_promotion_state']
+        if PREFERRED_API_VERSION in api_version:
+            pods_fp = array.list_pods(failover_preference=True, pending_only=True)
+            for pod in range(0, len(pods_fp)):
+                acpod = pods_fp[pod]['name']
+                pods_info[acpod]['failover_preference'] = pods_fp[pod]['failover_preference']
+                pods_info[acpod]['time_remainig'] = pods_fp[pod]['time_remainig']
```
Comment (dnix101) [inline collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py:876]: Suggest to change `pods_fp[pod]['time_remainig'] ` to `pods_fp[pod].get('time_remainig')` for backwards compatibility with purity 5.x.

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py [hunk 3]
```diff
@@ -767,10 +890,20 @@ def generate_pods_dict(array):
                 'mediator': pods[pod]['mediator'],
                 'mediator_version': pods[pod]['mediator_version'],
             }
-        pods_fp = array.list_pods(failover_preference=True)
-        for pod in range(0, len(pods_fp)):
-            acpod = pods_fp[pod]['name']
-            pods_info[acpod]['failover_preference'] = pods_fp[pod]['failover_preference']
+            if pods_info[acpod]['arrays'][0]['frozen_at']:
```
Comment (dnix101) [inline collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py:893]: Suggest to change pods_info[acpod]['arrays'][0]['frozen_at'] to pods_info[acpod]['arrays'][0].get('frozen_at') for backwards compatibility with purity 5.x.

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py [hunk 4]
```diff
@@ -754,6 +826,57 @@ def generate_pgroups_dict(array):
     return pgroups_info
 
 
+def generate_rl_dict(array):
+    rl_info = {}
+    api_version = array._list_available_rest_versions()
+    if ACTIVE_DR_API in api_version:
+        rlinks = array.list_pod_replica_links()
```
Comment (dnix101) [inline collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py:833]: This line returns an error if the `purestorage` sdk version is lower than 1.19. After an upgrade to 1.19 it's fine.

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py (hunk 3 above)
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py]: Moved to a better location where version check is applied

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py (hunk 1 above)
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py]: Moved to a better location where version check is applied

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py (hunk 2 above)
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py]: actually a typo and in the wrong place

File: collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py (hunk 4 above)
Comment (sdodsley) [reply collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py]: Added warning message

**Pull request merged**